apps/07jquery-tic-tac-toe/test.js

```diff
@@ -0,0 +1,134 @@
+var jsdom = require("jsdom");
+var assert = require('assert');
+
+// send browser log statements to node console
+var virtualConsole = jsdom.createVirtualConsole().sendTo(console);
+
+// handy globals for assertions in each test
+var document;
+var $;
+
+// get jsdom ready before each test (and wait for document ready event)
+beforeEach(function (done) {
+    jsdom.env(
+        {
+            file: './apps/07jquery-tic-tac-toe/index.html',
+            scripts: [
+                "../vendor/jquery.js",
+                "./script.js"
+            ],
+            virtualConsole: virtualConsole,
+            done: function (err, window) {
+                if (err) throw err;
+                document = window.document;
+                $ = window.$;
+                $(document).on('ready', function () {
+                    done();
+                });
+            }
+        }
+    );
+});
+
+describe('board state', function () {
+    it('should start empty', function () {
+        for (var i = 0; i <= 8; i += 1) {
+            assert.equal($('div[data-cell=' + i + ']').text(), '');
+        }
+    });
+
+    it ('should put an x then an o on alternating clicks', function () {
+        var $cell1 = $('div[data-cell=1]');
+        var $cell2 = $('div[data-cell=2]');
+        $cell1.click();
+        assert.equal($cell1.text(), 'X');
+        $cell2.click();
+        assert.equal($cell2.text(), 'O');
+    });
+});
+
+describe('detect a win', function () {
+    describe('horizontal', function () {
+        it('should detect a win in row 1', function () {
+            [ { x: 0, o: 3 }, { x: 1, o: 4 }, { x: 2, o: 6 }].forEach(function (cellNumbers) {
+                $('div[data-cell=' + cellNumbers.x + ']').click();
+                $('div[data-cell=' + cellNumbers.o + ']').click();
+            });
+            assert.equal($('#announce-winner').text(), 'Player X won!');
```

Comment:
Overall, this is great. My concern is that these tests are tied to the contents of `$('#announce-winner')`.

Now, [the app guide](http://intermediate.austincodingacademy.com/_book/app_guide_seven.html) the winning text should be `'Player ' + playerTurn + 'Wins!'`.

But what makes this more difficult is not this minor difference. I'm generally concerned that students will be off on their win text a little bit.

Could we change these tests so that they call `horizontalWin()`, `verticalWin()`, `diagonalWin()`, and `checkForWin()` directly, @tjefferson08?